### ISSUE TITLE
✅ 🐞 Bug Report: Styles don't always update until manual Regenerate CSS - [ED-10594]

### ISSUE THREAD

**Issue by pingram3541** [elementor/elementor#4464]
- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

## Description

When updating an element's style settings, i.e. background color, font size...custom css, anything really and saving, it doesn't always update on front end.  In fact inspecting the code on the front end, one can still see the post-XXX.css file still has old css rules from before the edits.  After trying every cache busting trick in the book, only manually going into the back end: **Elementor > Tools > Regenerate CSS** does the trick.

## Steps to reproduce

This is not something I have been able to easily reproduce and have experienced it on multiple sites but it doesn't happen every time on every site.  More of a once in a while kind of thing.  However, when I have run into it, it was pretty clear that the css did not update until immediately after performing the regenerate action.  It's happened enough times to know I was not just getting lucky with some other cache mechanism finally clearing.

## Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a default WordPress theme active.
- [ ] I can reproduce this bug consistently using the steps above.

## Environment (not a specific site but all very lite and many no plugins except E + Epro)

<details>
<summary>System Info</summary>
WP 4.9.6<br>
E 2.12<br>
Epro 2.06<br>
Hello Theme (unmodified)<br>
- No hosting cache, no Cloudflare cache, no cache plugins, no wp-config defined cache, no mu cache plugins, no google account cache
</details>

**Comment by obe1968** [comment]
Have you tried turning off your cache plugin as well as an server cache and then update your css? You should never have a cache plugin active while designing or doing updates.

**Comment by bainternet** [comment]
@pingram3541 

Please contact our support at support @elementor.com for guidance in resolving this issue.

**Comment by taropaa** [comment]
I can also verify that this is an issue. Sometimes in Multisite environments, we can't always disable caching. Also when clients ask for updates, the idea is to be able to deliver them quicker not have to go through extra steps when saving. I am surprised Elementor has no built in cache busting on save. It should save the css rules as a post-XXX-v2.css etc. Every time I have to go into the backend and clear the cache after I change a font size, and then clear my browser cache, flush Cloudflare feed my cat and change my socks. So even if I perform a design change, the client will still yell at me for not implementing the change... This is a pretty serious oversight. Divi re-generates those caches and truly is WYSIWYG. (using both commercially, currently considering switching to Elementor).
Now imagine throwing a CDN in the mix which has a 15 min expiry time, it would be maddening editing an elementor site under those conditions.

**Comment by heyfletch** [comment]
Curious if anyone has found a good cache-busting technique for Elementor's CSS files. I have the same problems described here.

**Comment by sigersmit** [comment]
I am having the same problem. Chrome debugger gives a 404 for the Elementor Stylesheets. Regenerating works, but, managing over 25 sites, this is a lot of extra work.

**Comment by GabiArzu** [comment]
Anyone managed to solve that problem? I'm having it with 2 different sites and different themes.

**Comment by heyfletch** [comment]
There's no solution to the fundamental problem yet. Elementor's css saving is not ideal because it does not cache-bust css files, which really should be the standard practice. Using the following work-arounds, I've been able to reduce the problems:

1. Enable inline CSS while developing
2. Use Autoptimize, which pulls in the static css files and creates a single cache-busted css file
3. Turn off page caching on the server
4. Don't use Cloudflare caching
5. Have a handy bookmark to the Regenerate page, and all other Clear Cache options (Autoptimize, Cloudflare, Host's, etc.)

I may update this list from time to time. It sucks that we have to go through all this trouble when the fix (css cache-busting) is a common practice given all the ways css gets cached (server, browser, etc.).

**Comment by pingram3541** [comment]
@heyfletch  Cache busting can be such a pain.  Adding a few more to your list:

1. If using Chrome, tick settings option to disable browser cache while Dev Tools is open
2. Log out of Google account (if logged in to Chrome), it has been known to also cache website content that doesn't release even when choosing specifically to empty the cache and perform a hard reload

**Comment by renehamburger** [comment]
I've also come across this now, which can lead to truly broken layouts if the html is updated but not the css!
I was really surprised to see that the fingerprinted css file names like `/wp-content/cache/minify/69845.css` did not change even though their content changed! Is this an Elementor or a WordPress bug??

**Comment by heyfletch** [comment]
> Adding a few more to your list:

Thanks @pingram3541! I added them to my list.

**Comment by heyfletch** [comment]
@renehamburger this raises another reason why caching plugins are not a real solution to Elementor cache busting. In order to work, the caching plugin needs to be compatible to listen for all Elementor changes, and then bust the cache. Autoptimize does this, but I believe there are a few settings it does not "hear", e.g., Elementor settings changes.

**Comment by renehamburger** [comment]
@heyfletch, you're right, automatic cache busting would be the ideal solution. I can live with doing this manually for now. In the case of W3 Total Cache, though, even the _manual_ purging of the cache couldn't resolved it, which is very surprising and an absolute show stopper.

**Comment by bentalgad** [comment]
Please fix it! even thought it won't make you more money :)

**Comment by acornavi** [comment]
Same problem from time to time. Please, make this work smoothly. It is very distracting and damaging for the work.

**Comment by leginnosivad** [comment]
I also get this problem quite a lot when pushing from staging to live versions of the site. I thought it was something to do with the 3rd part theme we had built our site on, so I rebuilt it from scratch with the Elementor theme, but I still get the same issue happening. 

I quite often have to regenerate the CSS to get it to display correctly

**Comment by marekmaurizio** [comment]
And now Regenerate it's gone.

**Comment by ridinghoodmedia** [comment]
I have several cache problems with Elementor, been having them for several months. Right now the inline styles from a en elementor shortcode template (inserted into a style tag by Elementor normally) shows up in the Elementor editor but not on the frontend. Regenerated CSS, busted Elementor's cache, nothing. Still debugging.

**Comment by pixelicous** [comment]
Amazing, look at the amount of comments but not even one response from the elementor team.

I am pulling out from their paid subscription and moving away as i can from them as soon as i can

**Comment by webenadam** [comment]
you should move to oxygen it is by far better from my last 2 months experience..

**Comment by johodesign** [comment]
I have the same issue unfortunately, and the client is crazy mad and I do not have a solution for it.

**Comment by bentalgad** [comment]
Stop working with elementor :) it's a very unstable plugin.

‫בתאריך יום א׳, 1 בנוב׳ 2020 ב-1:31 מאת ‪johodesign‬‏ <‪
notifications@github.com‬‏>:‬

**Comment by johodesign** [comment]
Okay, but it is a built website... I cannot change that in a day. 
And you are using what? Oxygen? Or?

**Comment by webenadam** [comment]
Yes I get it, but there is no easy solution. try to deactivate and
activate, make big changes to the pages or so...

I usually don't use a builder at all (much faster, much more stable, and
much more flexible), but when a client ask for it i use elementor / oxygen
/ beaver builder and so...



[image: BENΛDΛM | בניית אתרים] <https://benadam.co.il/>


‫בתאריך יום א׳, 1 בנוב׳ 2020 ב-1:54 מאת ‪johodesign‬‏ <‪
notifications@github.com‬‏>:‬

**Comment by ErikRobles** [comment]
This is January 26th 2021 and this is still an issue. Every single change I make, no matter how minor, I have to regenerate css in Elementor tools. Some have suggested turning off the caching plugin while designing but some hosts (shared that is) will give you a query limit exceeded and your site will suddenly stop working. This has happened to me more than once. Is there any solution on Elementor's part. I am using pro and am finding my productivity is hampered by always going to the tools setting to regenerat css. Thank you.

**Comment by heyfletch** [comment]
@ErikRobles 2 suggestions: 1) when designing, try switching CSS to the inline mode 2) open a new github issue. Elementor rarely if ever reacts to closed to issues.

**Comment by marekmaurizio** [comment]
Elementor support: this is not support go to github.
Elementor github: have you tried opening a ticket?
Repeat.

**Comment by Iconicdigitalw** [comment]
Seems this issue never really went off. It's getting more annoying by the day. I do not have any cache plugin yet, everytime I make any small change on one elementor page, other pages have issues... Please provide a lasting solution!!! I cannot be regenerating CSS all the time!!!

**Comment by Iconicdigitalw** [comment]
Seems this issue never really went off. It's getting more annoying by the day. I do not have any cache plugin yet, everytime I make any small change on one elementor page, other pages have issues... Please provide a lasting solution!!! I cannot be regenerating CSS all the time!!!

**Comment by dmquin** [comment]
This is happening right now, not in all sites, but some... i have to regenerate css on simple changes and i'm not using any cache plugin and not catching in private window...

**Comment by acornavi** [comment]
I found that the button color set in the Global settings doesn't apply as
the button background color default.
Also, returning to History, typography changes containing global fonts
don't show after Undo.

**Comment by nadworks** [comment]
So glad I found this thread. Thought I was going mad.
Same problem - on some random sites, every change requires the **_tools > regenerate..._** action. It's absolutely ludicrous and my clients are seriously frustrated. I have no cache plugin running at all while I develop the sites. 
Why is this not being dealt with?

**Comment by hunterbit** [comment]
I have same problem. it's very dangerous for business.

**Comment by ofmarconi** [comment]
### **I have same problem**

**Comment by ofmarconi** [comment]
Isn't it possible for a Snippet to always perform CSS regeneration every time something is saved in Elementor? @bainternet

**Comment by acornavi** [comment]
It is possible and easy to implement, but Elementor doesn't make correct
moves and doesn't listen to users.

**Comment by ofmarconi** [comment]
Hello everybody!

Support helped me and explained that 99% of cases are related to Optimization Plugins or "Additional Breakpoints"

In my case disabling the additional breakpoints has solved this problem so far.

Something I wouldn't imagine, as Elementor recommends leaving this function enabled, but in my case this has resolved.

Thank you and I hope to help

**Comment by Iconicdigitalw** [comment]
I found a mystery solution yesternight.I regenerated css, changed Elementor’s Hello theme to Astra and I never had the issue again.It seems the issue is something that has to do with Hello theme. Is everyone who has the issue also using Hello theme?   Iconic Digital World is a registered LLC in the United States.Iconicdigitalworld.com...functional, creative, attractive solutions…Webdesign | Content Writing | Consulting From: Orlando MarconiSent: Sunday, July 17, 2022 3:08 PMTo: elementor/elementorCc: Iconicdigitalw; CommentSubject: Re: [elementor/elementor] Styles don't always update until manual Regenerate CSS (#4464) Hello everybody!Support helped me and explained that 99% of cases are related to Optimization Plugins or "Additional Breakpoints"In my case disabling the additional breakpoints has solved this problem so far.Something I wouldn't imagine, as Elementor recommends leaving this function enabled, but in my case this has resolved.Thank you and I hope to help—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

**Comment by acornavi** [comment]
It is not an optimization plugin since you will have the same problem with
all the optimization turned off. Regarding the "additional break points,"
it's Elementor functionality, so devs can make it work properly or
depreciate it and not use it. Half-baked products are not an excuse but the
product of bad coding practice.

**Comment by ofmarconi** [comment]
@acornavi you are absolutely right!

@bainternet has been aware of this error since 16 May 2018, and since then has only closed this ticket.

This and other silly mistakes that make Elementor a joke to the competition.

**Comment by johodesign** [comment]
@ofmarconi
What competition do you use? 

Because I have not found the right alternative yet. I have tried oxygen... but it is not good for some purposes. do use bricks or what? Ireally need user friendliness.

**Comment by ofmarconi** [comment]
@johodesign I think it's still not worth other solutions if you're used to this one, but these small mistakes make me question whether I should invest in getting to know others better.

**Comment by El-shamar** [comment]
How am I still experiencing this issue in 2022? It's really frustrating

**Comment by cwebmaster** [comment]
I still face that issue we are on Aug 22 2022 and never fixed. You guys focusing on how to add new features that are useless and keep bugs giving users hard time for years, this is scary, we are really tired of all this...

**Comment by threadi** [comment]
Happens to me now in an Elementor installation of several dozen. Also no caching plugin active. I always have to call Regenerate CSS manually to see changes properly.

**Comment by threadi** [comment]
Yes, I saw that one, but that doesn't apply to the project here. Elementor Pro support has so far only advised me to clear the browser cache, which is not actually true. I'll stay tuned.

**Comment by threadi** [comment]
Update: it was the custom breakpoints after all. A closer test only with them confirmed the suspicion. I guess in the hectic rush the day before yesterday I hadn't noticed that.

**Comment by marowi85** [comment]
We have encountered this issue ourselves on a completely new website we have recently created.

@bainternet Why is this issue closed, although it still exists?

**Comment by PieterBrink** [comment]
SO wierd we have this across all our sites 
Elementor updates -> breaks the css on the site until I can "regenerate CSS" I can't - 

I wonder if there is a way one can trigger the "Regenerate CSS" on Elementor update - or use the WP upgrader_process_complete hook - could save some time wasted each time the plugin is updated - I am not skilled enough to write it thought.

**Comment by ofmarconi** [comment]
@PieterBrink did you see that in my case i solved it? have you tried this?

> In my case, disabling the additional breakpoints solved this problem so far.

**Comment by PieterBrink** [comment]
@ofmarconi thanks for reply - I can look to try to emulate this thanks - but to my knowledge it happens on our simpler sites too where we don't have extra break points. A lot of our sites have an extra break point for design.. would not be possible to disable for all.  hence us wondering if there is a way to trigger "Regenerate CSS" programmatically.

**Comment by ofmarconi** [comment]
@PieterBrink I know how frustrating this is! Had to abandon this feature because of this stupid bug.

But if you find a way to Regenerate the CSS let me know, it would help me in solving other things too.

**Comment by Anil-Chaudhary-netclues** [comment]
Hello @elementor team, We are facing same issue, when we edit page and change anything then front side all the mess up and not show CSS changes. once i am go to Tool-> Elementor -> Regenerate then i will show the changes. But this is very hard to do for my client. Please let me know how to fix them. Please suggest/help me about this issue.

Thanks you in advance your QUICK help.

**Comment by Dixfractions** [comment]
Problem still exists as of Today. Seems like Elementor likes to come up with new features but they never fix everlasting bugs.

**Comment by JonathanCrossland** [comment]
@elementor We are still getting this problem in 2023. Please resolve this or provide a satisfactory work around.

**Comment by taskymonk** [comment]
I think it can be solved by adding a cron job to regenerate it on specific events or at some frequency.
Maybe this could solve the problem temporarily.

@elementor please look into the matter and fix the issue.

**Comment by mjunaidit** [comment]
So glad I found this thread. I had the same issue with 2 of my sites and it happened when I had to use Shortcode within tabs, and it only affected the mobile view - Where the columns rather than stacking on top of each other, would still show side by side like on desktop. Whenever I made a change to the Elementor Template that I had added a shortcode for in tabs, this happened. I've disabled additional breakpoints under features for now on one site. Will see if that works.

**Comment by sinanisler** [comment]
still happening.

**Comment by sinanisler** [comment]
are you sure ? made any tests ?
 @Sarathkumar311 

thank you

**Comment by jackladrera** [comment]
> It is coming from Elementor -> Settings -> Features -> Additional Custom Breakpoints.
> 
> Make Inactive this setting - Additional Custom Breakpoints.
> 
> it will work fine.
> 
> ![image](https://user-images.githubusercontent.com/6386642/227937735-94a4c529-4488-4395-8955-80ec3601bbef.png)

this helps! thank you

**Comment by sinanisler** [comment]
yep disabling additional break points fixes this bug.

**Comment by andreasjhagen** [comment]
> It is coming from Elementor -> Settings -> Features -> Additional Custom Breakpoints.
> 
> Make Inactive this setting - Additional Custom Breakpoints.
> 
> it will work fine.
> 
> ![image](https://user-images.githubusercontent.com/6386642/227937735-94a4c529-4488-4395-8955-80ec3601bbef.png)

Nice catch, this appears to be the cause. Hope we can get an official fix for this

**Comment by opticalresolution** [comment]
Sorry but I don't think that disabling custom breakpoints fixes the issue... 

I had really high hopes however I ran a test and specifically disabled this feature on 3 sites I manage, meanwhile leaving it default on 3 other sites and it made absolutely no difference. 

EVERYTIME elementor updates the entire css of the site breaks and I have to go in and click "Regenerate CSS Files & Data" and then clear the Cache. It's gotten to the point where I'm literally checking all the sites I manage daily to avoid this EMBARRASSMENT and even sometime it still catches me. 

No, it's not my Cache plugin. (Using full version WP-Rocket on some and WP Fastest Cache on others). Have also tested disabling all cache plugins and the CSS still breaks everytime elementor has an update. 

![Screen Shot 2023-05-29 at 8 20 18 AM](https://github.com/elementor/elementor/assets/93783672/04257a76-b211-4db1-8562-cd96db54a936)
![Screen Shot 2023-05-29 at 8 19 55 AM](https://github.com/elementor/elementor/assets/93783672/03f50b1e-4160-4d5a-8d11-b2555eb03f29)
![Screen Shot 2023-05-29 at 8 20 10 AM](https://github.com/elementor/elementor/assets/93783672/6135b35f-ee58-403d-8260-ef4d3da6cc6f)
![Screen Shot 2023-05-29 at 8 21 59 AM](https://github.com/elementor/elementor/assets/93783672/d8717b9b-dd6c-4725-8b34-ecdd2afdece5)

It does NOT seem to break on sites I made before switching to using the Hello Elementor theme. (Like Astra or Ocean WP) 

So is it a problem with Elementor updating, or with using the Hello Elementor theme? Not sure yet. 

Message to the Elementor team, this is a very known issue. I mean this post was started May 15, 2018..... Why are we not taking steps to solve the problem? I love that you're innovating all the time with things like the megamenu and custom loop features, etc, but when the CSS is breaking every other day at random times throughout the night, none of that really matters does it? 

I will be reaching out to Elementor across other platforms until I can get an answer on this because frankly it's embarrassing.

**Comment by mcrowell** [comment]
@opticalresolution > Sorry but I don't think that disabling custom breakpoints fixes the issue...

I can also confirm that the custom breakpoints isn't always the problem. I can't figure out what causes this bug.

**Comment by opticalresolution** [comment]
Woke up today and checked the sites, only 1 broken today but its one that I clicked "Regenerate CSS Files & Data" yesterday morning on. Speaking to someone right now at Elementor and of course getting the usual run-around. Had to be escalated to Tier 2 support which will continue by email. 

I did want to post something I hadn't noticed before until now going back and forth with this guy. When the CSS breaks it breaks in a few specific places. The header, the footer, anywhere there is an icon list, and also the post pages all break. 

Opened the console and sure enough: 

![Screen Shot 2023-05-30 at 7 30 41 AM copy](https://github.com/elementor/elementor/assets/93783672/99621829-29f1-4f41-84d1-2bd57fe97d0d)

widget-icon-list
widget-nav-menu
widget-posts

And of course after I regenerate, no more console errors. 

![Screen Shot 2023-05-30 at 7 52 10 AM](https://github.com/elementor/elementor/assets/93783672/83bf1b3a-10c4-4aee-b83f-7b77829670a0)

This still all means nothing, but just wanted to post incase anyone else can confirm the same behavior.

Also is there a way to open this issue thread back up?

**Comment by larsaam** [comment]
@bainternet You closed this issue, but it is not fixed..

**Comment by Dixfractions** [comment]
I also have the issue with the Astra theme

**Comment by sinanisler** [comment]
just disable the **additional break points** for now until we get a fix for this.

**Comment by opticalresolution** [comment]
@sinaster it's not that simple my friend. As stated above that does not fix the issue. I have it disabled on all the sites and still having problems. What theme are you running and what versions of elementor/elementor pro?

**Comment by sinanisler** [comment]
hello-elementor with latest pro

**Comment by opticalresolution** [comment]
UPDATE: 

tldr: Elementor > Settings > Features (tab) Set these features to disabled and **save**: 
OPTIMIZED DOM OUTPUT
IMPROVED ASSET LOADING
IMPROVED CSS LOADING (most likely the problem)
ACCESSABILITY IMPROVEMENTS

**Click Save**

___

I think I figured out where the problem is coming from. 

I have been experimenting with a handful of my own sites split testing settings and comparing. This morning one of them was broken but the others were not so I checked Elementor > Settings > Features (tab) and found that 4 settings under the "Stable Features" header were set to ACTIVE, but disabled on my other sites which did NOT have broken CSS. 

So I disabled them like my other sites and clicked save.

![DISABLE THESE TO FIX CSS](https://github.com/elementor/elementor/assets/93783672/3fdb67e0-de8d-4753-adab-66fa2cd4ffea)

Note that until now, the only solution to fix the broken css was to click the Regenerate CSS Files & Data button. You could refresh the page a million times, clear the cache, nothing. You HAD TO click that button. 

I disabled the 4 "stable" features above and clicked save. 

The broken page was open in a separate incognito window (not logged into WP) and i refreshed the page. The CSS was perfect once again. Zero issues. WTF ELEMENTOR.....

I never once had to click Regenerate CSS Files & Data. 

Now I'm not entirely sure which of the 4 settings is to blame, but I can assume it's "IMPROVED CSS LOADING". Ironically enough does the exact opposite of what it claims to do. I will probably try and narrow it down to 1 later next week but for now I'll just be monitoring everything and seeing if it breaks again. For now, all 4 are staying DISABLED. 

Also, my page speed insights score did not change at all after disabling these features so nothing to worry about there. 

Some of these features are not stable and I would recommend disabling anything you are not using. Did have pretty good luck so far with the mega menu and loop features, so some things are ok.

Here are my exact settings I'm now running. 

![elementor-settings](https://github.com/elementor/elementor/assets/93783672/49309ce0-b228-4139-8843-5617ba478553)

Hope really hope this helps some people out there who have been struggling like I have with this bug. 

Would love to hear if it works for anyone else.

**Comment by sharjeelkhanvmi** [comment]
Try This Code

//Auto Clear REGENERATE CACHE OF ELEMENTOR
add_action('save_post', 'clear_elementor_cache');

function clear_elementor_cache() {
  // Check if Elementor is loaded and the hook has fired
  if ( did_action( 'elementor/loaded' ) ) {
  // Automatically purge and regenerate the Elementor CSS cache
    \Elementor\Plugin::instance()->files_manager->clear_cache();
  }
}
//Auto Clear REGENERATE CACHE OF ELEMENTOR


Link to Stack overflow for more details
https://stackoverflow.com/questions/75499137/search-wordpress-hook-for-completed-elementor-update
https://stackoverflow.com/questions/75388444/elementor-templates-breaking-page-layout-is-there-any-way-to-fix-this-issue-exc

**Comment by Audiovoyeur** [comment]
Thanks for this @sharjeelkhanvmi!  This is totally whatt I was thinking. 

So If we do something like this:
* Check if Elementor is loaded
* Automatically purge and regenerate the Elementor CSS cache
* Attach the function to 'save_post' action
* Attach the function to 'upgrader_process_complete' action


!!!!! WARNING UNTESTED CODE - only EXAMPLE only !!!!!

`// Function to clear Elementor cache
function clear_elementor_cache() {
    // Check if Elementor is loaded
    if ( did_action( 'elementor/loaded' ) ) {
        // Automatically purge and regenerate the Elementor CSS cache
        \Elementor\Plugin::instance()->files_manager->clear_cache();
    }
}

// Attach the function to 'save_post' action
add_action('save_post', 'clear_elementor_cache');

// Attach the function to 'upgrader_process_complete' action
add_action('upgrader_process_complete', 'clear_elementor_cache',10, 2);

function check_plugin_update( $upgrader_object, $options ) {
    if ($options['action'] == 'update' && $options['type'] == 'plugin' ) {
        foreach($options['plugins'] as $each_plugin) {
            if ($each_plugin == 'elementor/elementor.php') {
                clear_elementor_cache();
            }
        }
    }
}
add_action('upgrader_process_complete', 'check_plugin_update', 10, 2);
`
!!!!! WARNING UNTESTED CODE - EXAMPLE only !!!!

it would be great if the code could refresh CSS the once eliminator is undated - as this is when (my) issue happens. 

So two questions would be - 
How to affect the same to eliminator PRO plugin

**Comment by JonathanCrossland** [comment]
Hi All, its a long thread, I did not read, but to share a workaround.

`
//php to clear cache
  $elementor = \Elementor\Plugin::$instance;
  $elementor->files_manager->clear_cache(true);
  
  
  //you can add this to a hook in wordpress or behind a querystring, or call it directly whenever
  
   if( isset($_GET['runcache']) && 'true' === $_GET['runcache'] ) {
        $elementor = \Elementor\Plugin::$instance;
        $elementor->files_manager->clear_cache(true);
        }
  `

**Comment by PurpleWebCreative** [comment]
I have just carried out @opticalresolution suggestion and i had a screwed site loaded in an incognito window. Once i had made these changes i went back to the incognito session and then simply refreshed the window and the site loaded. No need to regen CSS. Surely this has to be dealt with by support. Its been such an issue to us we are even trialing breakdance

**Comment by opticalresolution** [comment]
@PurpleWebCreative I'm glad to hear this worked for you! Since I changed my settings, I have had no CSS issues with any sites I manage. You may be in the clear my friend.

**Comment by amisha-ncompass** [comment]
@opticalresolution i made those changes but it didnt work

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Core 3.15.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂

**Comment by opticalresolution** [comment]
@PurpleWebCreative ^^^ replying to you from the other thread. My problems were fixed for a while but then came back ofcourse after the changes were reverted. Sorry for not getting to you in time on the other post.

**Comment by nicholaszein** [comment]
Hello peeps! 👋 

We just [released a new patch](https://github.com/orgs/elementor/discussions/23925#:~:text=Responsive%20Template%20Designs) for this issue in **version 3.17-beta**, and would love to have your feedback!

How is it working for you?
Did you find any issues when editing templates in responsive modes?
Is it working correctly even with cache turned on?

We want to make sure we are covering all bases here to avoid further conflicts, and we would really appreciate your feedback!

As always, please remember to make a backup before updating, and if possible test on a staging environment.

Let us know how is it working for you. 🙏

**Comment by opticalresolution** [comment]
Hey @nicholaszein I'll keep a close eye on it and let you know if I have any issues moving into next week. Appreciate you listening to all the user feedback and taking action. Great job to the Elementor team 🙌🏼

**Comment by nicholaszein** [comment]
Hi @opticalresolution!

Thank you. We really appreciate your help! I already got confirmation about other issues being solved by this fix. Knowing this one is also fixed on your end would really help! 🙂

We made a lot of tests with this one, but every site is unique, so we want to make sure everyone is happy! 😊

**Comment by nwolfdigital** [comment]
Please let us know when this release is came out from beta phase. We working on a complex webshop and this problem always comes up. The pain point here is that our woocommerce product grid dissapears on mobile view from time to time (buyers not able to shop on the page!) and we are not always on the situation to click on the Regenerate CSS button. For example ehile we sleep And  (Also on our responsive blog these things happen)
We really need this problem to be fixed.

**Comment by nicholaszein** [comment]
Hello everybody!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor v3.17! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
